src/kaltura-player.js

```diff
@@ -54,6 +53,7 @@ class KalturaPlayer extends FakeEventTarget {
 
   setMedia(mediaConfig: ProviderMediaConfigObject): void {
     this._logger.debug('setMedia', mediaConfig);
+    this.reset();
```

Comment:
can't move this here - expectiation is that once loadMedia request is done the player will reset itself immediately and not keep playing until API server responds.
If you call this API via the playlist manager then it should call reset by itself.